src/sgx-verifier/SGXVerifier.sol

```diff
@@ -0,0 +1,220 @@
+     * Public View Functions *
+     *************************/
+
+    /// @notice Return the EIP712 hash for `ProveBundleSignatureData`
+    /// @dev The struct is below, also see https://github.com/scroll-tech/sgx-prover/blob/main/crates/rpc/src/types.rs
+    /// ```text
+    /// struct ProveBundleSignatureData {
+    ///     layer2ChainId uint64;
+    ///     numBatches uint32;
+    ///     prevStateRoot bytes32;
+    ///     prevBatchHash bytes32;
+    ///     postStateRoot bytes32;
+    ///     batchHash bytes32;
+    ///     postWithdrawRoot bytes32;
+    /// }
+    /// ```
+    function getProveBundleSignatureDataHash(
+        uint64 layer2ChainId,
+        uint32 numBatches,
+        bytes32 prevStateRoot,
+        bytes32 prevBatchHash,
+        bytes32 postStateRoot,
+        bytes32 batchHash,
+        bytes32 postWithdrawRoot
+    ) public view returns (bytes32) {
+        bytes32 structHash = keccak256(
+            abi.encode(
+                _BUNDLE_PAYLOAD_TYPEHASH,
+                layer2ChainId,
+                numBatches,
+                prevStateRoot,
+                prevBatchHash,
+                postStateRoot,
+                batchHash,
+                postWithdrawRoot
+            )
+        );
+        return _hashTypedDataV4(structHash);
+    }
+
+    /// @inheritdoc IZkEvmVerifierV2
+    ///
+    /// @dev Encoding for `publicInput`
+    /// ```text
+    /// | layer2ChainId | numBatches | prevStateRoot | prevBatchHash | postStateRoot | batchHash | postWithdrawRoot |
+    /// |    8 bytes    |  4  bytes  |   32  bytes   |   32  bytes   |   32  bytes   | 32  bytes |     32 bytes     |
+    /// ```
+    ///
+    /// This function will revert when the proof is invalid.
+    function verify(bytes calldata bundleProof, bytes calldata publicInput) external view override {
+        uint64 layer2ChainId;
+        uint32 numBatches;
+        bytes32 prevStateRoot;
+        bytes32 prevBatchHash;
+        bytes32 postStateRoot;
+        bytes32 batchHash;
+        bytes32 postWithdrawRoot;
+        assembly {
+            layer2ChainId := shr(192, calldataload(publicInput.offset))
+            numBatches := shr(224, calldataload(add(publicInput.offset, 0x08)))
+            prevStateRoot := calldataload(add(publicInput.offset, 0xc))
+            prevBatchHash := calldataload(add(publicInput.offset, 0x2c))
+            postStateRoot := calldataload(add(publicInput.offset, 0x4c))
+            batchHash := calldataload(add(publicInput.offset, 0x6c))
+            postWithdrawRoot := calldataload(add(publicInput.offset, 0x8c))
+        }
+
+        bytes32 hash = getProveBundleSignatureDataHash(
+            layer2ChainId,
+            numBatches,
+            prevStateRoot,
+            prevBatchHash,
+            postStateRoot,
+            batchHash,
+            postWithdrawRoot
+        );
+        address prover = ECDSA.recover(hash, bundleProof);
+
+        if (attestedProvers[prover].validUntil < block.timestamp) {
```

Comment:
So in this version, any registered prover can submit an SGX proof, right?

Reply:
Yeah, we allow any registered prover submit an SGX proof in current version. This will make the contracts code clean.